### ISSUE TITLE
Problems with `defaults` and `constants`

### ISSUE THREAD

**Issue by smathot** [esdalmaijer/PyGaze#13]
# The problem

Right now, configuration options are stored in the module `defaults` and overwritten by values from the module `constants` if available. This works fine when you indeed use a `constants.py` file to set options. But if you try to change options at runtime things get rough.

Basically, what happens is that all modules start like this:

``` .python
from pygaze.defaults import *
try:
    from constants import *
except:
    pass
```

What this does is copy the values into the module itself. This means that changing an option in defaults, like so ....

``` .python
from pygaze import defaults
defaults.DISPTYPE = 'opensesame'
```

... _only_ works when no modules have been loaded yet. If a module has already been loaded, it needs to explicitly reloaded in order for the option to take effect. For the plug-ins, this required a hack, which is implemented in `pygaze_init.reload_pygaze()`.
- https://github.com/esdalmaijer/PyGaze/blob/master/opensesame_plugins/pygaze_init/pygaze_init.py#L83

Another problem is that you can specify many options through keywords as well as the config system. When you actually start using the keywords, things easily get messy and confusing.
# A solution

In general terms, there should be a config system in which all options are stored in a single place, without any copying of config options. (Copying now happens implicitly due to the import structure shown above.)

There are many ways to do this, but one way would be to mimick the config system of OpenSesame.
- https://github.com/smathot/OpenSesame/blob/master/libqtopensesame/misc/config.py

Basically, there is a single instance of the `config` class, called `cfg` (a singleton design pattern), and this can be accessed from anywhere in the code, like so:

``` .python
from libqtopensesame.misc.config import cfg
cfg.my_option = 'a value' # Set the option
print cfg # Get the option
```

Perhaps the `constants.py` file could be read and then used to set options through this config system on start-up.

I would also do away with all the keyword arguments, such as `disptype`, that should really be set through the config system.

What do you think?
